### ISSUE TITLE
The content returned by torch.nn.Sequential.append is inconsistent with the official

### ISSUE THREAD

**Issue by ChengYen-Tang** [dotnet/TorchSharp#852]
According to the documentation, append returns a Sequential. But TorchSharp is void
https://pytorch.org/docs/stable/generated/torch.nn.Sequential.html#torch.nn.Sequential.append

**Comment by ChengYen-Tang** [comment]
Another question
Why does this place use IList instead of ICollection, the package that my project depends on will output Array
Now, I need toList, and then TorchSharp toArray(). Feels like there's a lot of performance loss in data copying

https://github.com/dotnet/TorchSharp/blob/c55e9f535aa46c4dfa104d839019553b693616c0/src/TorchSharp/Tensor/Tensor.Factories.cs#L140

**Comment by NiklasGustafsson** [comment]
> According to the documentation, append returns a Sequential. But TorchSharp is void

Thanks for pointing that out. It will be fixed in the next version.

> Why does this place use IList instead of ICollection, the package that my project depends on will output Array

ICollection seems wrong as the abstract interface here, since it does not imply order, which IList<T> does.

**Comment by ChengYen-Tang** [comment]
1. I have a different idea, ICollection has nothing to do with the order, the developer decides which collection to use will affect the order.
2. Then why not use T[], but use IList.
3. I think that for machine learning frameworks, performance is the most important, and other limitations should be increased without affecting performance. My input is 500M data, and it will waste a lot of time in processing ToList() and ToArray, and at the same time increase the pressure of GC (Note: ArrayPool can be used in some places)

**Comment by lostmsu** [comment]
Ideally there should be an overload taking `ReadOnlySpan<T>`

**Comment by NiklasGustafsson** [comment]
If you're starting with an array, I would recommend using `tensor` instead of `as_tensor()` -- the array overloads are already available there.

**Comment by ChengYen-Tang** [comment]
![image](https://user-images.githubusercontent.com/32535494/204685705-2ff3afc6-53e7-4f33-887f-f20e54360b28.png)
![image](https://user-images.githubusercontent.com/32535494/204685775-44bb4a44-55c4-4ae7-9f21-01fc9a94f41a.png)

But I can't insert array.

**Comment by NiklasGustafsson** [comment]
@ChengYen-Tang I can reproduce that with this:

```C#
            float[] a = new float[12];
            var t = torch.as_tensor(a);
```

I'm troubleshooting, but in the meantime, you can replace it with 'frombuffer' or cast the array to IList<float> in the call (for now).

```C#
            float[] a = new float[12];
            var t = torch.as_tensor((IList<float>)a);
```

**Comment by NiklasGustafsson** [comment]
The reason there's an ambiguity is that there's an implicit operator that converts an array to a Tensor. You should be able to pass the array to the 'forward' directly, unless you need to reshape it or change the element type or device. 

For example:

```C#
            float[] a = new float[12];
            var x = ReLU();
            var t = x.forward(a);
```

**Comment by NiklasGustafsson** [comment]
That still leaves the ambiguity in place, which I should find a way to address. I'll leave this issue open to track.

**Comment by NiklasGustafsson** [comment]
You can also use `torch.tensor()` if you want to make sure that a copy is being made. `as_tensor()` will only do that when necessary because of the `dtype` or `device` arguments.

**Comment by ChengYen-Tang** [comment]
> @ChengYen-Tang I can reproduce that with this:
> 
> ```cs
>             float[] a = new float[12];
>             var t = torch.as_tensor(a);
> ```
> 
> I'm troubleshooting, but in the meantime, you can replace it with 'frombuffer' or cast the array to IList in the call (for now).
> 
> ```cs
>             float[] a = new float[12];
>             var t = torch.as_tensor((IList<float>)a);
> ```

I don't know, what happened.
this can work
![image](https://user-images.githubusercontent.com/32535494/204695843-20d81680-bdd5-4f0b-a7f4-2f5e5e73506b.png)

Can anyone tell me why ?😂
![image](https://user-images.githubusercontent.com/32535494/204696156-40e04914-bb30-49ec-ae12-27d8784171ca.png)

This project is a library

**Comment by ChengYen-Tang** [comment]
> 存在歧義的原因是有一個隱式運算符將陣列轉換為張量。您應該能夠直接將數位傳遞給“轉發”，除非您需要重塑它或更改元素類型或設備。
> 
> 例如：
> 
> ```cs
>             float[] a = new float[12];
>             var x = ReLU();
>             var t = x.forward(a);
> ```

Since I'm duplicating a python project, I want to preserve features as much as possible

**Comment by NiklasGustafsson** [comment]
@ChengYen-Tang -- what happened is that you changed the static type from float[] to Array, for which there is a separate overload of 'as_tensor()'

**Comment by NiklasGustafsson** [comment]
> 存在歧義的原因是有一個隱式運算符將陣列轉換為張量。您應該能夠直接將數位傳遞給“轉發”，除非您需要重塑它或更改元素類型或設備。

Unfortunately, I am not going to be able to respond to any commentary or questions unless they're in English. Sorry.

**Comment by ChengYen-Tang** [comment]
Ok, it looks like T[] and Array are not object oriented inheritance.

**Comment by NiklasGustafsson** [comment]
I'm going to add overloads for 1-D arrays, which are the ones that are affected by the ambiguity. N-D arrays (N > 1) will be caught by the already existing 'Array' overload.

**Comment by NiklasGustafsson** [comment]
I just released 0.99.1 to NuGet. The array overloads are in that release.
